### PR TITLE
Avoid SDK crash if the state changes since the last authentication

### PR DESCRIPTION
**Pull request by gorandostanic**

### CODE REVIEW

File: connect/src/com/telenor/connect/utils/ConnectUtils.java [hunk 1]
```diff
@@ -25,12 +25,16 @@ public static void parseAuthCode(String callbackUrl, ConnectCallback callback) {
         Validator.notNullOrEmpty(callbackUrl, "callbackUrl");
 
         Uri uri = Uri.parse(callbackUrl);
-        Validator.validateAuthenticationState(uri.getQueryParameter("state"));
+        if (!Validator.validState(uri.getQueryParameter("state"))) {
```
Comment (atanasdb) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: are you sure u've included all the files in the commit

File: connect/src/com/telenor/connect/utils/ConnectUtils.java [hunk 2]
```diff
@@ -49,4 +53,11 @@ public static void sendTokenStateChanged(boolean state) {
         intent.putExtra(LOGIN_STATE, state);
         localBroadcastManager.sendBroadcast(intent);
     }
+
+    private static Map<String, String> getErrorParams(String error, String errorDescription) {
```
Comment (atanasdb) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: I'm not sure this abstraction is worth it. It's one indirection more, and getErrorParams is used in one place only anyway.

File: connect/src/com/telenor/connect/utils/ConnectUtils.java (hunk 1 above)
Comment (gorandostanic) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: What do you mean?

File: connect/src/com/telenor/connect/utils/ConnectUtils.java (hunk 2 above)
Comment (gorandostanic) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: Done!

File: connect/src/com/telenor/connect/utils/ConnectUtils.java (hunk 1 above)
Comment (atanasdb) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: Nothing ;).
I guess you will want to delete validateAuthenticationState now as it's not used any more.

File: connect/src/com/telenor/connect/utils/ConnectUtils.java (hunk 2 above)
Comment (atanasdb) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: My github diff got a bit cluttered. You are actually calling getErrorParams twice. So maybe it make sense to leave it as it was the in the first commit. I'm sorry about that.

File: connect/src/com/telenor/connect/utils/ConnectUtils.java (hunk 1 above)
Comment (gorandostanic) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: I'll do it in another patch (that I am currently working on)

File: connect/src/com/telenor/connect/utils/ConnectUtils.java (hunk 2 above)
Comment (gorandostanic) [inline connect/src/com/telenor/connect/utils/ConnectUtils.java]: I think I’ll leave it as it is, since it's in the spirit of this repo :)

**Pull request merged**